argo/src/Argo/ServerState.hs

```diff
@@ -122,24 +123,35 @@ nextAppState ::
   appState {-^ The new application state -} ->
   IO StateID
 nextAppState server prevStateID newAppState = do
-  destroyAppState server prevStateID
+  destroyAppState' server prevStateID
   uuid <- UUID.nextRandom
   modifyIORef' (serverStatePool server) $ HM.insert uuid newAppState
   return $ StateID uuid
 
+-- | Desctroy an app state so it is no longer available for requests.
+destroyAppState' ::
```

Comment:
If I understand correctly, the only difference between `destroyAppState` and `destroyAppState`' is that the latter will only destroy an app state if the ID is not initial? If so, it would be worth documenting this.

Reply:
It's slightly more complicated... so I added a bunch of comments regarding these various scary effectful functions: https://github.com/GaloisInc/argo/pull/174/commits/4513fda6daaf5c075bfb1a16948cea4cec38e111